### PR TITLE
Update authority-key-id when preparing pre-issued pre-cert

### PR DESCRIPTION
**Pull request by daviddrysdale**

### CODE REVIEW

**Comment by daviddrysdale** [comment]
For #39

File: trillian/integration/ct_integration.go [hunk 1]
```diff
@@ -651,12 +652,13 @@ func buildNewPrecertData(cert, issuer *x509.Certificate, signer crypto.Signer) (
...
 // buildLeafTBS builds the raw pre-cert data (a DER-encoded TBSCertificate) that is included
 // in the log.
-func buildLeafTBS(precertData []byte, issuerName *pkix.Name) ([]byte, error) {
+func buildLeafTBS(precertData []byte, issuer *x509.Certificate) ([]byte, error) {
 	reparsed, err := x509.ParseCertificate(precertData)
 	if err != nil {
 		return nil, fmt.Errorf("failed to re-parse created precertificate: %v", err)
 	}
-	return x509.BuildPrecertTBS(reparsed.RawTBSCertificate, issuerName)
+	result, err := x509.BuildPrecertTBS(reparsed.RawTBSCertificate, issuer)
```
Comment (AlCutter) [inline trillian/integration/ct_integration.go]: Why not just return the values from the call as before?

File: trillian/integration/ct_integration.go [hunk 2]
```diff
@@ -682,6 +684,9 @@ func makePreIssuerPrecertChain(chain []ct.ASN1Cert, issuer *x509.Certificate, si
 	preIssuer.PublicKeyAlgorithm = x509.ECDSA
 	preIssuer.PublicKey = preSigner.Public()
 	preIssuer.ExtKeyUsage = append(preIssuer.ExtKeyUsage, x509.ExtKeyUsageCertificateTransparency)
+	// Set a different subject-key-id for the intermediate.  Note that this needs to be the DER-encoded
+	// SEQUENCE, not just the key ID.
+	preIssuer.SubjectKeyId = []byte{0xde, 0xad, 0xbe, 0xaf}
```
Comment (AlCutter) [inline trillian/integration/ct_integration.go]: Why these particular numbers?

Comment (daviddrysdale) [inline trillian/integration/ct_integration.go]: They just need to be different than the subject ID of the real issuer (which is 0x01020304)

File: trillian/integration/ct_integration.go (hunk 1 above)
Comment (daviddrysdale) [inline trillian/integration/ct_integration.go]: Ah, that was left over from when I had some printfs in there.

File: trillian/integration/ct_integration.go (hunk 2 above)
Comment (daviddrysdale) [inline trillian/integration/ct_integration.go]: The comment is wrong though...

**Comment by daviddrysdale** [comment]
Added a comment and shifted to use a random subject-key-id for the special intermediate.

**Pull request merged**